### PR TITLE
Pagination handling + internal tweaks

### PR DESCRIPTION
**Pull request by JakubFrejlach**
This PR brings several improvements:

- new helper methods for pagination handling (CLOSES OSIDB-267)
- new method for obtaining fields from each model (internal usage for [griffon](https://github.com/RedHatProductSecurity/griffon))
- new exposed dictionary which describes query parameters for each endpoint module (internal usage for [griffon](https://github.com/RedHatProductSecurity/griffon))

### CODE REVIEW

File: osidb_bindings/templates/endpoint_module.py.jinja [hunk 1]
```diff
@@ -111,4 +111,12 @@ def sync(
     return sync_detailed(
         {{ kwargs(endpoint) }}
     ).parsed
+
+QUERY_PARAMS = {
```
Comment (JimFuller-RedHat) [inline osidb_bindings/templates/endpoint_module.py.jinja:115]: can we add this to tutorial and changelog as well ?

**Review by JimFuller-RedHat — APPROVED** [review]
nice and clean ... LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+### Added
+- added `next()`, `prev()` and `iterator()` methods for paginated
+  responses to make page browsing easier
```
[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply CHANGELOG.md:16]: Currently, this is only exposed on low level in experimental way for bindings, so imho no need to add that to the tutorial. We can make use of it and add it as some session operation, to just list the reosource fields and types. Wdyt?

File: osidb_bindings/templates/endpoint_module.py.jinja (hunk 1 above)
Comment (JakubFrejlach) [reply osidb_bindings/templates/endpoint_module.py.jinja:115]: We technically can, but this is the part of the low level bindings, I would rather expose that in some nice way via the session maybe or some helper on the surface so the user does not need to go deep inside the bindings logic.

File: CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JimFuller-RedHat) [reply CHANGELOG.md:16]: sounds reasonable

File: osidb_bindings/templates/endpoint_module.py.jinja (hunk 1 above)
Comment (JimFuller-RedHat) [reply osidb_bindings/templates/endpoint_module.py.jinja:115]: fine with me ;)

**Pull request merged**